### ISSUE TITLE
Double Square Bracket – somewhere – stop Doxygen

### ISSUE THREAD

**Issue by traud** [doxygen/doxygen#8914]
```
#include <stdio.h>

/*!
 * \brief Hello World!
 */
static void hello() {
	puts("[[");
}

/*!
 * \brief main
 * \retval 0 Always
 * \retval 0 Never
 */
int main(void) {
	hello();
	return 0;
}
```
should give “return value '0' of main has multiple documentation sections” when I use
```
EXTRACT_ALL   = YES
WARN_AS_ERROR = YES
```
as configuration. However, the above gives no error/warning at all.

**Analysis**:
The problem are the two opening square quotes. For example, if I write
a) `puts("[[]]");` closing both quotes 
b) `puts("[");` just a single open quote
c) `puts("[\0x5b");` one as hexadezimal
d) `puts("[\133");` one as octal
instead, all these four alternatives work and Doxygen gives the error/warning as expected. For me, looks like Doxygen simply quits parsing (or be more precise: ignores) the remainder of the document when *somewhere* in the document two consecutive _opening_ square quotes are found. Even worse, the following two examples do not work as expected either:
2. `puts("[[]");`
3. `puts("[[Hello]World]!");`
Consequently, to close such an ‘exceptional’ section, I have to place two closing square brackets consecutively (or avoiding two consecutive opening square brackets).

**Version**:
1.9.1 on Ubuntu 21.10, re-tested with current binary release (1.9.2)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8915

**Comment by traud** [comment]
I cannot comment on the proposed Pull Request (I do not understand the commit message), nevertheless, many thanks for taking over!

**Comment by albert-github** [comment]
Maybe you can build an master version with the proposed patch or use one of the "artifacts"  from https://github.com/doxygen/doxygen/actions/runs/1508040340 otherwise you have to wait to till it is integrated into an official release.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
